core/app/models/workarea/pricing/request.rb

```diff
@@ -48,9 +48,10 @@ def shippings
       def payment
         return unless @persisted_payment.present?
 
-        @payment ||= @persisted_payment.clone.tap do |payment|
-          payment.id = @persisted_payment.id # Ensure this isn't persisted
-        end
+        @payment ||=
+          Workarea::Payment.instantiate(@persisted_payment.as_document).tap do |payment|
+            payment.id = @persisted_payment.id # Ensure this isn't persisted
```

Comment:
I realized this actually still returns a savable document, this change will fix that:
```suggestion
            payment.new_record = true # Ensure this isn't persisted by raising duplicate key if saved
```

Reply:
Wouldn't that allow it to save?